services/src/pro/datasets/in_memory.rs

```diff
@@ -409,7 +514,10 @@ impl
     ) -> Result<Box<dyn MetaData<OgrSourceDataset, VectorResultDescriptor, VectorQueryRectangle>>>
     {
         ensure!(
-            self.dataset_permissions
+            self.backend
+                .read()
+                .await
+                .dataset_permissions
```

Comment:
and here

Reply:
fccf47a2459c22ceb51d2ac1d1b4e0237d13cfd1